lib/request.js

```diff
@@ -769,7 +769,9 @@ AWS.Request.addPromisesToClass = function addPromisesToClass(PromiseDependency)
         if (resp.error) {
           reject(resp.error);
         } else {
-          resolve(resp.data);
+          var data = AWS.util.copy(resp.data);
```

Comment:
Do we need to copy `resp.data` here? I'm a little worried about copying Buffers for operations that return binary data. Since you can't mix promises/callbacks/streams, we shouldn't really have to worry about mutating the `data` object.